halo2_proofs/src/plonk/lookup.rs

```diff
@@ -1,10 +1,17 @@
-use super::circuit::Expression;
+use super::circuit::{Expression, ExpressionMid};
 use ff::Field;
 use std::fmt::{self, Debug};
 
 pub(crate) mod prover;
 pub(crate) mod verifier;
 
+#[derive(Clone, Debug)]
+pub struct ArgumentV2<F: Field> {
+    pub(crate) name: String,
+    pub(crate) input_expressions: Vec<ExpressionMid<F>>,
+    pub(crate) table_expressions: Vec<ExpressionMid<F>>,
```

Comment:
Why `ExpressionMid`? Can you elaborate?

Reply:
`ExpressionMid` doesn't contain query indices.  This type is used here to remove the burden from the frontend of tracking query indices.